Packages/com.reese.nav/Agent/Systems/NavCollisionSystem.cs

```diff
@@ -0,0 +1,92 @@
+            var isDebugging      = IsDebugging;
+
+            Entities
+                .WithNone<NavProblem>()
+                .WithNone<NavPlanning, NavJumping, NavFalling>()
+                .WithAll<NavWalking, NavFlocking, LocalToParent>()
+                .WithReadOnly(physicsWorld)
+                .ForEach((Entity entity, int entityInQueryIndex, ref Translation translation, ref NavAgent agent, ref NavSteering steering,
+                    in LocalToWorld localToWorld, in Rotation rotation, in Parent surface) =>
+                {
+                    var averageHitDirection = float3.zero;
+                    var clostestHitDistance = agent.ObstacleAversionDistance;
+                    int hits = 0;
+                    
+                    // Cast 17 rays in a (flockingSettings.CollisionCastingAngle * 2) deg angle with the entities forward direction
+                    for (int i = 0; i < 17; ++i)
+                    {
+                        var rayModifier = Quaternion.AngleAxis((i / ((float)17 - 1)) * flockingSettings.CollisionCastingAngle * 2 - flockingSettings.CollisionCastingAngle, localToWorld.Up);
+                        var rayDirection = rotation.Value * rayModifier * Vector3.forward;
+                        
+                        if(isDebugging) 
+                            Debug.DrawRay(localToWorld.Position, rayDirection * agent.ObstacleAversionDistance, Color.cyan);
+
+                        var ray = new RaycastInput
+                        {
+                            Start = localToWorld.Position,
+                            End = localToWorld.Position + (float3)rayDirection * agent.ObstacleAversionDistance, 
+                            Filter = new CollisionFilter()
+                            {
+                                BelongsTo = NavUtil.ToBitMask(settings.ColliderLayer),
+                                CollidesWith = NavUtil.ToBitMask(settings.ObstacleLayer),
+                            }
+                        };
+                        
+                        if (physicsWorld.CastRay(ray, out RaycastHit hit))
+                        {
+                            var distance     = Vector3.Distance(localToWorld.Position, hit.Position);
+                            
+                            clostestHitDistance  = math.select(clostestHitDistance, distance, distance < clostestHitDistance);
+                            averageHitDirection += localToWorld.Position - hit.Position;
+                            
+                            hits++;
+                            
+                            if(isDebugging) 
+                                Debug.DrawLine(localToWorld.Position, hit.Position, Color.red);
+
+                        }
+                    }
+
+                    averageHitDirection /= hits;
+                    
+                    var scaler = 1 - (clostestHitDistance / agent.ObstacleAversionDistance);
```

Comment:
I would rename `scaler` to `scalar`. Also, the parens are unnecessary. The division will be performed first, then the subtraction, and that's without the parens.